x-pack/plugins/security_solution/public/common/store/sourcerer/helpers.ts

```diff
@@ -6,49 +6,117 @@
+import { TimelineEventsType } from '../../../../common';
+import { DEFAULT_DATA_VIEW_ID } from '../../../../common/constants';
+import { SelectedDataViewPayload } from './actions';
 
 export interface Args {
   eventType?: TimelineEventsType;
   id: SourcererScopeName;
   selectedPatterns: string[];
   state: SourcererModel;
 }
-export const createDefaultIndexPatterns = ({ eventType, id, selectedPatterns, state }: Args) => {
-  const kibanaIndexPatterns = state.kibanaIndexPatterns.map((kip) => kip.title);
-  const newSelectedPatterns = selectedPatterns.filter(
-    (sp) =>
-      state.configIndexPatterns.includes(sp) ||
-      kibanaIndexPatterns.includes(sp) ||
-      (!isEmpty(state.signalIndexName) && state.signalIndexName === sp)
-  );
-  if (isEmpty(newSelectedPatterns)) {
-    let defaultIndexPatterns = state.configIndexPatterns;
-    if (id === SourcererScopeName.timeline && isEmpty(newSelectedPatterns)) {
-      defaultIndexPatterns = defaultIndexPatternByEventType({ state, eventType });
-    } else if (id === SourcererScopeName.detections && isEmpty(newSelectedPatterns)) {
-      defaultIndexPatterns = [state.signalIndexName ?? ''];
+
+export const getScopePatternListSelection = (
+  theDataView: KibanaDataView | undefined,
+  sourcererScope: SourcererScopeName,
+  signalIndexName: SourcererModel['signalIndexName']
+): string[] => {
+  let patternList: string[] =
+    theDataView != null && theDataView.id !== null ? theDataView.patternList : [];
+
+  // when our SIEM DATA_VIEW is set, here are the defaults
+  if (theDataView && theDataView.id === DEFAULT_DATA_VIEW_ID) {
+    if (sourcererScope === SourcererScopeName.default) {
+      // indexOf instead of === because the dataView version of signals index
+      // will have a wildcard and the signalIndexName does not include the wildcard
+      patternList = patternList.filter((index) => index.indexOf(`${signalIndexName}`) === -1);
+    } else if (sourcererScope === SourcererScopeName.detections) {
+      patternList = signalIndexName != null ? [signalIndexName] : []; // set to signalIndexName whether or not it exists yet in the patternList
+    } else if (sourcererScope === SourcererScopeName.timeline) {
+      patternList =
+        signalIndexName != null
+          ? [
+              // indexOf instead of === because the dataView version of signals index
+              // will have a wildcard and the signalIndexName does not include the wildcard
+              // remove signalIndexName in case its already in there and add it whether or not it exists yet in the patternList
+              ...patternList.filter((index) => index.indexOf(`${signalIndexName}`) === -1),
+              signalIndexName,
+            ]
+          : patternList;
     }
-    return defaultIndexPatterns;
   }
-  return newSelectedPatterns;
+  return patternList.sort();
+};
+
+export const validateSelectedPatterns = (
```

Comment:
why the double !! in front of the `!!dataView?.patternList.some`?

Reply:
I took the parts i liked, let me know what you think of the new version

Reply:
> why the double !! in front of the `!!dataView?.patternList.some`?

Since `dataView?.` can return undefined the `!!` converts that to false. It might not be needed though 🤔